### PR TITLE
Optimize `invert` of `Scalar`

### PR DESCRIPTION
**Pull request by moCello**
Resolves: #78

### CODE REVIEW

File: src/scalar.rs [hunk 1]
```diff
@@ -624,9 +624,12 @@ impl Scalar {
         res
     }
 
-    /// Computes the multiplicative inverse of this element,
-    /// failing if the element is zero.
+    /// Computes the multiplicative inverse of this element, failing if the
+    /// element is zero.
     pub fn invert(&self) -> CtOption<Self> {
+        if *self == Scalar::one() || *self == -Scalar::one() {
+            return CtOption::new(*self, !self.ct_eq(&Self::zero()));
```
Comment (vlopes11) [inline src/scalar.rs:631]: Why do we check if `self.ct_eq(&Self::zero())`, considering we already know the scalar is either `-1` or `1`

Comment (vlopes11) [inline src/scalar.rs:631]: We need to keep in mind that this is no longer a constant time operation. My suggestion:

1) change the signature to `fn invert(&self) -> Option<Self>`
2) Create a new function `fn invert_ct(&self) -> CtOption<Self>` that will have the previous implementation

**Review by vlopes11 — CHANGES_REQUESTED** [review]

File: src/scalar.rs (hunk 1 above)
Comment (moCello) [reply src/scalar.rs:631]: For no reason, will change it to `true`

File: src/scalar.rs [hunk 2]
```diff
@@ -624,9 +624,18 @@ impl Scalar {
         res
     }
 
-    /// Computes the multiplicative inverse of this element,
+    /// Computes the multiplicative inverse of this element, failing if the
+    /// element is zero.
+    pub fn invert(&self) -> Option<Self> {
+        if *self == Scalar::one() || *self == -Scalar::one() {
```
Comment (vlopes11) [inline src/scalar.rs:630]: Maybe we cover also `0 -> None`?

**Review by vlopes11 — APPROVED** [review]
LGTM, optional nit

File: src/scalar.rs (hunk 2 above)
Comment (moCello) [reply src/scalar.rs:630]: Sure, why not

**Pull request merged**